akka-http-core/src/main/scala/akka/http/javadsl/ConnectHttp.scala

```diff
@@ -29,59 +29,89 @@ object ConnectHttp {
 
   // TODO may be optimised a bit to avoid parsing the Uri entirely for the known port cases
 
-  /** Extracts host data from given Uri. */
-  def toHost(uriHost: Uri): ConnectHttp = {
-    val s = uriHost.scheme.toLowerCase(Locale.ROOT)
-    if (s == "https") new ConnectHttpsImpl(uriHost.host.address, effectivePort(s, uriHost.port))
-    else new ConnectHttpImpl(uriHost.host.address, effectivePort(s, uriHost.port))
-  }
+  /** Extracts HTTP or HTTPS connection data from given Uri. */
+  def toHost(uriHost: Uri): ConnectHttp =
+    toHost(uriHost, uriHost.port)
 
-  def toHost(host: String): ConnectHttp = {
-    if (isHttpOrHttps(host)) toHost(Uri.create(host))
-    else toHost(Uri.create(s"http://$host"))
-  }
+  /**
+   * Extract HTTP or HTTPS connection data from given host.
+   *
+   * The host string may contain a URI or a <host>:<port> pair.
+   */
+  def toHost(host: String): ConnectHttp =
+    toHost(createUriWithScheme("http", host))
 
+  /**
+   * Extracts HTTP or HTTPS connection data from given host and port.
+   *
+   * The host string may contain a URI or a <host>:<port> pair. In both cases the
```

Comment:
Thanks for documenting this. I think the API should be improved. I created https://github.com/akka/akka-http/issues/833 to track this.